neuraxle/pipeline.py

```diff
@@ -486,3 +490,222 @@ def should_resume(self, data_container: DataContainer) -> bool:
                 return True
 
         return False
+
+
+"""
+Idea for checkpoints : 
+
+    The streaming pipeline algorithm could go find the optional checkpoint step for each sub pipeline.
+    In the future, a ResumableStreamingPipeline that extends this class should exist to support checkpoints.
+    
+    The Barrier should ideally join the data so that the ram does not blow up (iterable, lazy loading, cache ??)
+    Maybe we can implement a LazyLoadingDataContainer/CachedDataContainer class or something that could be returned.
+    
+    pipeline = Pipeline([
+        MiniBatchSequentialPipeline([
+
+            A(),
+            B(),
+            Barrier(joiner=Joiner()),
+            [Checkpoint()]
+
+            C(),
+            D(),
+            Barrier(joiner=Joiner()),
+            [Checkpoint()]
+
+        ]),
+        Model()
+    ])
+    
+    pipeline = Pipeline([
+        MiniBatchSequentialPipeline([
+            ParallelWrapper([
+                NonFittableA(),
+                NonFittableB(),
+            ])
+            Barrier(joiner=Joiner()),
+            [Checkpoint()]
+
+            C(),
+            D(),
+            Barrier(joiner=Joiner()),
+            [Checkpoint()]
+
+        ]),
+        Model()
+    ])
+"""
+
+
+class Barrier(NonFittableMixin, NonTransformableMixin, BaseStep):
+    pass
+
+
+class MiniBatchSequentialPipeline(NonFittableMixin, Pipeline):
+    """
+    Mini Batch Sequential Pipeline class to create a pipeline processing data inputs in batch.
+    """
+
+    def __init__(self, steps: NamedTupleList, batch_size):
+        Pipeline.__init__(self, steps)
+        self.batch_size = batch_size
+
+    def transform(self, data_inputs: Any):
+        """
+        :param data_inputs: the data input to transform
+        :return: transformed data inputs
+        """
+        self.setup()
+
+        current_ids = self.hash(
+            current_ids=None,
+            hyperparameters=self.hyperparams,
+            data_inputs=data_inputs
+        )
```

Comment:
Extract method for 

        current_ids = self.hash(
            current_ids=None,
            hyperparameters=self.hyperparams,
            data_inputs=data_inputs
        )

Reply:
(e.g.: duplicated code in fit_transform)

Reply:
Not reseolved.